players/base.py

```diff
@@ -357,7 +357,11 @@ def has_overdrive(self, combat=False) -> bool:
     def is_turn(self) -> bool:
         return memory.main.get_battle_char_turn() == self.id
 
-    def in_danger(self, danger_threshold, combat=False) -> bool:
+    def in_danger(self, danger_threshold) -> bool:
+        if memory.main.battle_active():
```

Comment:
could make it `combat = memory.main.battle_active()` for more compact code.

Reply:
Updated, please check again.